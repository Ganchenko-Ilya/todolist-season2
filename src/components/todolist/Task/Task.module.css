.wrapperCheckbox img {
  cursor: pointer;
  width: 23px;
  height: 23px;
}
.taskIsDone {
  opacity: 0.5;
}
.wrapperCheckbox {
  display: flex;
  align-items: center;
}

/* .wrapperTasks {
  display: flex;
  justify-content: center;
  margin: 5px 0;
} */
.wrapperTasks  li {
  list-style-type: none;
  display: flex;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 5px;
}
.wrapperTasks  li:last-child {
  margin: 0;
}
li span {
  font-size: 20px;
  margin-right: 5px;
}