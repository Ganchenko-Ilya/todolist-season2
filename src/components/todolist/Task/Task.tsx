import { Checkbox } from "@mui/material";
import DeleteIcon from "@mui/icons-material/Delete";
import { EditableSpan } from "../../../otherComponents/editableSpan/EditableSpan";
import s from "./Task.module.css";
import React, { ChangeEvent, useCallback } from "react";

import { ChangeTaskResponseType, TasksStatuses } from "../../../api/todolists-api";
import { StatusType } from "../../../store/app-reducer";

export const Task = React.memo((props: TaskTypeProps) => {
  console.log(`Task`);

  const { deleteTask, editTitle, onChangeStatus, statusTodo } = props;
  const { id, status, title, statusLoad } = props.el;

  const editTitleHandlerTask = useCallback(
    (newTitle: string) => {
      editTitle(id, newTitle);
    },
    [editTitle]
  );
  const onChangeStatusHandler = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      onChangeStatus(id, e.currentTarget.checked ? TasksStatuses.Complited : TasksStatuses.New);
    },
    [onChangeStatus]
  );
  const onCLickDeleteTaskHanlder = useCallback(() => {
    deleteTask(id);
  }, [deleteTask]);

  return (
    <div className={s.wrapperTasks}>
      <li className={status ? s.taskIsDone : ""}>
        <EditableSpan
          statusTodo={statusTodo}
          statusLoad={statusLoad}
          title={title}
          editTitle={editTitleHandlerTask}
        />
        <div className={s.wrapperCheckbox}>
          <Checkbox
            sx={{ width: "0px", height: "18px" }}
            onChange={onChangeStatusHandler}
            size="small"
            checked={!!status}
            disabled={statusTodo === "loading" || statusLoad === "loading"}
          />

          <DeleteIcon
            className={s.deleteIcon}
            style={
              statusTodo === "loading" || statusLoad === "loading"
                ? { pointerEvents: "none", opacity: 0.5, cursor: "pointer" }
                : {}
            }
            onClick={onCLickDeleteTaskHanlder}
          />
        </div>
      </li>
    </div>
  );
});

export type TaskTypeProps = {
  el: ChangeTaskResponseType;
  statusTodo: StatusType;
  editTitle: (id: string, newTitle: string) => void;
  onChangeStatus: (id: string, status: number) => void;
  deleteTask: (id: string) => void;
};
