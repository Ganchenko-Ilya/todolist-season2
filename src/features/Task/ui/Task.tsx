import { Checkbox } from "@mui/material";
import DeleteIcon from "@mui/icons-material/Delete";
import s from "./Task.module.css";
import React, { ChangeEvent, useCallback } from "react";
import { ChangeTaskResponseType, StatusType } from "shared/types";
import { TasksStatuses } from "shared/enum/enumApi";
import { EditableSpan } from "shared/ui";

export const Task = React.memo((props: TaskTypeProps) => {
  console.log(`Task`);

  const { deleteTask, editTitle, onChangeStatus, statusTodo } = props;
  const { id, status, title, statusLoad } = props.el;

  const editTitleHandlerTask = useCallback(
    (newTitle: string) => {
      editTitle(id, newTitle);
    },
    [editTitle, id],
  );
  const onChangeStatusHandler = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      onChangeStatus(id, e.currentTarget.checked ? TasksStatuses.Completed : TasksStatuses.New);
    },
    [onChangeStatus, id],
  );
  const onCLickDeleteTaskHanlder = useCallback(() => {
    deleteTask(id);
  }, [deleteTask, id]);

  return (
    <div className={s.wrapperTasks}>
      <li className={status ? s.taskIsDone : ""}>
        <EditableSpan statusTodo={statusTodo} statusLoad={statusLoad} title={title} editTitle={editTitleHandlerTask} />
        <div className={s.wrapperCheckbox}>
          <Checkbox
            sx={{ width: "0px", height: "18px" }}
            onChange={onChangeStatusHandler}
            size="small"
            checked={!!status}
            disabled={statusTodo === "loading" || statusLoad === "loading"}
          />

          <DeleteIcon
            className={s.deleteIcon}
            style={
              statusTodo === "loading" || statusLoad === "loading"
                ? { pointerEvents: "none", opacity: 0.5, cursor: "pointer" }
                : {}
            }
            onClick={onCLickDeleteTaskHanlder}
          />
        </div>
      </li>
    </div>
  );
});

export type TaskTypeProps = {
  el: ChangeTaskResponseType;
  statusTodo: StatusType;
  editTitle: (id: string, newTitle: string) => void;
  onChangeStatus: (id: string, status: number) => void;
  deleteTask: (id: string) => void;
};
