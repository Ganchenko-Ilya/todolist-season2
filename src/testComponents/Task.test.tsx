import renderer from "react-test-renderer";


import { Task } from "../components/todolist/Task/Task";


test("test code Components", () => {
  const result = renderer.create(<Task deleteTask={() => {}} editTitle={() => {}} el={{id:'1',isDone:false,title:'css'} } onChangeStatus={() => {}} />).toJSON();
  expect(result).toMatchSnapshot();
});