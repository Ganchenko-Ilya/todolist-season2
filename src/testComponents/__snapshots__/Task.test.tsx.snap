// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test code Components 1`] = `
<div
  className="wrapperTasks"
>
  <li
    className=""
  >
    <div
      className="editWrapper"
    >
      <span
        onDoubleClick={[Function]}
      >
        css
      </span>
    </div>
    <div
      className="wrapperCheckbox"
    >
      <span
        className="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeSmall css-1oxcmra-MuiButtonBase-root-MuiCheckbox-root"
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={null}
      >
        <input
          checked={false}
          className="PrivateSwitchBase-input css-1m9pwf3"
          data-indeterminate={false}
          onChange={[Function]}
          required={false}
          type="checkbox"
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="CheckBoxOutlineBlankIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
        />
      </svg>
    </div>
  </li>
</div>
`;
